hitokage-lua/src/widgets/clock.rs

```diff
@@ -25,23 +24,18 @@ impl UserData for ClockUserData {
     methods.add_method("get_type", |_, this, _: ()| Ok(this.r#type.clone()));
 
     methods.add_method("get_format", |_, this, _: ()| Ok(this.get_format()?));
-
     methods.add_method("set_format", |lua, this, value: mlua::Value| {
       Ok(this.set_format(lua, value)?)
     });
 
-    methods.add_meta_method(
+    methods.add_meta_method::<_, mlua::String, _>(
       "__index",
-      |lua, instance, value| -> Result<mlua::Value<'lua>, mlua::Error> {
-        match value {
-          Value::String(s) => match s.to_str()? {
-            "type" => Ok(lua.to_value(&instance.r#type.clone())?),
-            "format" => Ok(lua.to_value(&instance.get_format()?)?),
-            _ => Ok(Value::Nil),
-          },
+      |lua, instance, key| -> Result<mlua::Value<'lua>, mlua::Error> {
+        match key.to_str().unwrap() {
+          "type" => Ok(lua.to_value(&instance.r#type.clone())?),
           _ => Ok(Value::Nil),
         }
       },
-    )
+    );
```

Comment:
Why support this syntax at all? No need to store `r#type` as a struct member.

This is for greater discussion on supporting syntactic sugar such as `type`, `format`

I have am having trouble with using `methods.add_meta_method("__newindex", ...` which is holding back most of the work on syntactic sugar. Either convert this to `get_type` or a method rather than a "field"

If we remove it be sure to remove `r#type` declarations here

https://github.com/codyduong/hitokage/blob/0679370e6e1cfc2296e7fb497e99a5023c3abd8d/hitokage-lua/src/widgets/mod.rs#L18-L42

Reply:
Moved to #16 for greater discussion